test/test_remap.py

```diff
@@ -0,0 +1,124 @@
+import os
+import numpy as np
+import numpy.testing as nt
+import xarray as xr
+
+from unittest import TestCase
+from pathlib import Path
+
+import uxarray as ux
+
+from uxarray.core.dataset import UxDataset
+from uxarray.core.dataarray import UxDataArray
+
+from uxarray.remap import nearest_neighbor
+
+current_path = Path(os.path.dirname(os.path.realpath(__file__)))
+gridfile_ne30 = current_path / "meshfiles" / "ugrid" / "outCSne30" / "outCSne30.ug"
+gridfile_CSne30 = current_path / "meshfiles" / "ugrid" / "outCSne30" / "outCSne30.ug"
+dsfile_vortex_CSne30 = current_path / "meshfiles" / "ugrid" / "outCSne30" / "outCSne30_vortex.nc"
+gridfile_geoflow = current_path / "meshfiles" / "ugrid" / "geoflow-small" / "grid.nc"
+dsfile_v1_geoflow = current_path / "meshfiles" / "ugrid" / "geoflow-small" / "v1.nc"
+dsfile_v2_geoflow = current_path / "meshfiles" / "ugrid" / "geoflow-small" / "v2.nc"
+dsfile_v3_geoflow = current_path / "meshfiles" / "ugrid" / "geoflow-small" / "v3.nc"
+
+
+class TestNearestNeighborRemap(TestCase):
+    """Tests for nearest neighbor remapping."""
+
+    def test_remap_to_same_grid_corner_nodes(self):
```

Comment:
I believe we should only be concerned about (and providing) `uxds.nearest_neighbor_remap()` and `uxda.nearest_neighbor_remap()`, not public `uxarray.remap. nearest_neighbor()` that takes in data as type of `np.ndarray` (See related concerns in `nearest_neighbor.py`). Even if it's going to be user API, it should take in `UxDataarray`.